File: Arrays & Hashing/Two Sum.cpp
class Solution {
public:
  vector<int> twoSum(vector<int>& nums, int target) {
    map<int, int> mp;
    for (auto i : nums) {
      mp[i]++;
    }
    int temp, firstIndex, secondIndex;
    for (int i = 0; i < nums.size(); i++) {
      temp = target - nums[i];
      if (temp == nums[i] && mp[temp] < 2) {
        continue;
      }
      if (mp[temp]) {
        firstIndex = i;
        break;
      }
    }
    for (int i = 0; i < nums.size(); i++) {
      if (nums[i] == temp) {
          secondIndex = i;
      }
    }
    return {firstIndex, secondIndex};
  }
};

File: neetcode/Arrays & Hashing/Two-Sum.go
func twoSum(nums []int, target int) []int {
    freq := make(map[int]int)
    for idx, num := range nums {
        freq[num] = idx
    }
    for idx, num := range nums {
        if i, ok := freq[target - num]; ok && idx != i {
            return []int{idx, i}
        }
    }
    return nil
}